### PR TITLE
refactor(storage): remove v1 downgrades from integration test read paths

### LINKED ISSUE #7050: [Feature]: Upgrade e2e storage tests to use v2 Storage API and OTLP
### Requirement

When we test storage implementations in e2e tests we should be using v2 Storage API

### Problem

The existing e2e tests work off the v1 model which does not adequately represent some edge cases possible in OTLP data model, such as scope attributes.

https://github.com/jaegertracing/jaeger/blob/5873d1ffa8541446f62781eb6a87f0823bed6e57/internal/storage/integration/integration.go#L348

### Proposal

* [ ] Upgrade this write path to use v2 Storage API (TraceWriter)
* [ ] Remove unnecessary downgrades done in the tests themselves, leave the downgrades to the storage implementations (when they don't yet support v2 API)
* [ ] Incrementally swap test fixtures to be stored as OTLP JSON instead of v1 model JSON

### Open questions

_No response_

### PR DESCRIPTION
**Pull request by SoumyaRaikwar**
## Which problem is this PR solving?
- Resolves #7050 (Part 1/2)
- Removes v1 downgrades from integration test read paths to prepare for full OTLP migration (in progress)
- Fixes deprecated Elasticsearch client usage
- Fixes lint errors (depguard, context usage)

## Description of the changes
- Refactored `integration` package tests to use `t.Context()` instead of `context.Background()` for better test timeout handling.
- Replaced deprecated `IndexTemplateExists` and `IndexDeleteTemplate` in Elasticsearch tests with direct API calls.
- Implemented `CreateDependencyWriter` in Cassandra v2 factory (shim to v1) to support dependency storage tests.
- Fixed `goleak` issues in tests.
- Removed unused `nolint` directives.
- Minor fixes in `trace_compare.go` and `v1adapter/translator.go`.

## How was this change tested?
- All integration tests passing.
- `make lint-go` passing.

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality (N/A - refactoring existing tests)
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`

### CODE REVIEW

**Comment by SoumyaRaikwar** [comment]
@yurishkuro  could you please review my pr

File: internal/storage/integration/trace_compare_otlp.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+				span.References = []model.SpanRef{}
+			}
+		}
+		trace.Spans = append(trace.Spans, batch.Spans...)
+	}
+	return trace
+}
+
+// CollectOTLPTraces collects all traces from an iterator into a slice
+func CollectOTLPTraces(iterTraces iter.Seq2[[]ptrace.Traces, error]) ([]ptrace.Traces, error) {
```
Comment (yurishkuro) [inline internal/storage/integration/trace_compare_otlp.go]: please use existing functions

internal/storage/v2/v1adapter/translator.go
47:func V1TracesFromSeq2(otelSeq iter.Seq2[[]ptrace.Traces, error]) ([]*model.Trace, er

internal/jptrace/aggregator.go
16:func AggregateTraces(tracesSeq iter.Seq2[[]ptrace.Traces, error]) iter.Seq2[ptrace.Traces, error] {

internal/jiter/iter.go
11:func CollectWithErrors[V any](seq iter.Seq2[V, error]) ([]V, error) {
22:func FlattenWithErrors[V any](seq iter.Seq2[[]V, error]) ([]V, error) {

internal/jptrace/spaniter.go
20:func SpanIter(traces ptrace.Traces) iter.Seq2[SpanIterPos, ptrace.Span] {

File: internal/storage/integration/integration.go [hunk 2]
```diff
@@ -324,29 +333,35 @@ func (s *StorageIntegration) testGetTrace(t *testing.T) {
...
+			t.Logf("Error loading trace: %v", err)
 			return false
 		}
-		if len(traces) == 0 {
+		if len(otlpTraces) == 0 {
 			return false
 		}
-		actual = traces[0]
-		return len(actual.Spans) == len(expected.Spans)
+		actualOTLP = otlpTraces[0]
```
Comment (yurishkuro) [inline internal/storage/integration/integration.go]: please avoid unnecessary renaming - once this refactoring is finished we don't need to keep seeing "otlpTraces" when it's just "traces", so there's no need to introduce these names now either

File: internal/storage/integration/trace_compare_otlp.go (hunk 1 above)
Comment (SoumyaRaikwar) [reply internal/storage/integration/trace_compare_otlp.go]: Thanks for the pointers
I’ve updated the storage integration tests to reuse the shared helpers:

- `jiter.FlattenWithErrors` for collecting from `iter.Seq2`
- `GetFirstTrace` / `OTLPTracesToV1Slice` for converting to v1 right before comparison

Tests no longer define their own collector, and variable names are back to neutral (`traces` / `actual`).

File: internal/storage/integration/integration.go (hunk 2 above)
Comment (SoumyaRaikwar) [reply internal/storage/integration/integration.go]: Renamed back to `traces` / `actual` and now only convert to v1 right before comparison. No new `otlp*` names in the tests anymore.

**Comment by SoumyaRaikwar** [comment]
I’ve updated the storage integration tests to reuse the shared helpers:
Tests no longer define their own collector, and variable names are back to neutral (`traces` / `actual`).

@yurishkuro could you please review again

**Comment by SoumyaRaikwar** [comment]
@yurishkuro could you please review

File: internal/storage/integration/integration.go [hunk 3]
```diff
@@ -193,12 +195,13 @@ func (s *StorageIntegration) testGetServices(t *testing.T) {
 					StartTimeMin: time.Now().Add(-2 * time.Hour),
 					StartTimeMax: time.Now(),
 				})
-				traces, err := v1adapter.V1TracesFromSeq2(iterTraces)
+				traces, err := jiter.FlattenWithErrors(iterTraces)
```
Comment (yurishkuro) [inline internal/storage/integration/integration.go]: what is the point of this change? How is this better than calling `v1adapter.V1TracesFromSeq2`?

Comment (SoumyaRaikwar) [reply internal/storage/integration/integration.go]: Sorry for missing that—you're absolutely right. `V1TracesFromSeq2` is exactly what we need here; no point in doing the two-step conversion when we already have a helper that does it in one call. Updated.

**Comment by SoumyaRaikwar** [comment]
@yurishkuro could you please take another look

File: internal/storage/integration/integration.go [hunk 4]
```diff
@@ -229,10 +232,10 @@ func (s *StorageIntegration) helperTestGetTrace(
 	expected := s.writeLargeTraceWithDuplicateSpanIds(t, traceSize, duplicateCount)
 	expectedTraceID := v1adapter.FromV1TraceID(expected.Spans[0].TraceID)
 
-	actual := &model.Trace{} // no spans
+	var actual ptrace.Traces
 	found := s.waitForCondition(t, func(_ *testing.T) bool {
 		iterTraces := s.TraceReader.GetTraces(context.Background(), tracestore.GetTraceParams{TraceID: expectedTraceID})
-		traces, err := v1adapter.V1TracesFromSeq2(iterTraces)
+		traces, err := jiter.FlattenWithErrors(iterTraces)
```
Comment (yurishkuro) [inline internal/storage/integration/integration.go]: what is the point of this change? You still transform the result in to v1 model via GetFirstTrace().

**Comment by yurishkuro** [comment]
please address comments

**Comment by SoumyaRaikwar** [comment]
@yurishkuro 
I've updated the PR to use v1adapter.V1TracesFromSeq2 as requested, removing the custom OTLP helpers.

Note: To make this work safely for the integration tests, I had to fix a bug in internal/storage/v2/v1adapter/translator.go. It was performing a shallow copy of Process tags, which caused shared reference issues when multiple spans reused the same process. I added a deep copy to resolve this.

**Comment by SoumyaRaikwar** [comment]
Regarding Issue #7050

Upgrade Read Path: This PR (#7761) achieves this by consuming OTLP iterators directly from storage.
Upgrade Write Path: PR #7812 (@Manik2708) appears to address this by refactoring tests to write ptrace.Traces directly to the TraceWriter.
Remove Downgrades: Currently, we still downgrade to V1 for assertions.
Once #7812 done, it will functionally unblock the "Remove unnecessary downgrades" goal by allowing us to perform assertions directly on ptrace.Traces (preserving high-fidelity OTLP data like Scope attributes that V1 loses).

How would you like me to coordinate? I can pause till #7812 get merged , which seems to remain the critical piece for completing the full V2 transition described in the issue.

**Comment by yurishkuro** [comment]
do we still need this?

**Comment by Manik2708** [comment]
> do we still need this?

I think we can close this, as the issue reffered is closed
